packages/babel-plugin/src/utils/css-map.ts

```diff
@@ -0,0 +1,263 @@
+import { css, cssMap } from '@compiled/react';
+const borderStyleMap = cssMap({
+    none: { borderStyle: 'none' },
+    solid: { borderStyle: 'solid' },
+});
+const Component = ({ borderStyle }) => <div css={css(borderStyleMap[borderStyle])} />
+\`\`\`
+    `;
+};
+
+/**
+ * Retrieves the leftmost identity from a given expression.
+ *
+ * For example:
+ * Given a member expression "colors.primary.500", the function will return "colors".
+ *
+ * @param expression The expression to be evaluated.
+ * @returns {string} The leftmost identity in the expression.
+ */
+const findBindingIdentifier = (
+  expression: t.Expression | t.V8IntrinsicIdentifier
+): t.Identifier | undefined => {
+  if (t.isIdentifier(expression)) {
+    return expression;
+  } else if (t.isCallExpression(expression)) {
+    return findBindingIdentifier(expression.callee);
+  } else if (t.isMemberExpression(expression)) {
+    return findBindingIdentifier(expression.object);
+  }
+
+  return undefined;
+};
+
+/**
+ * Retrieves the key from a given expression or property from a given member expression.
+ *
+ * For example:
+ * Given a member expression `colors[variant]`, the function will return `variant` (as t.Identifier).
+ * Given a member expression `colors["variant"]`, the function will return `"variant"` (as t.StringLiteral).
+ * Given a member expression `colors.variant, the function will return `"variant"` (as t.StringLiteral).
+ * Given a member expression `colors.variant.nested, the function will return `undefined`.
+ * Given a object property `{ variant: 'something' }`, the function will return `"variant"` (as t.StringLiteral).
+ * Given a object property `{ "variant": 'something' }`, the function will return `"variant"` (as t.StringLiteral).
+ * Given a object property `{ [someDynamicVariable]: 'something' }`, the function will return `undefined`.
+ *
+ * @param {t.ObjectProperty | t.MemberExpression} The expression to be evaluated.
+ * @returns {t.Identifier | t.StringLiteral | undefined} The key or property from the expression.
+ */
+const getKeyOrProperty = (
+  value: t.ObjectProperty | t.MemberExpression
+): t.Identifier | t.StringLiteral | undefined => {
+  const keyOrProperty = t.isObjectProperty(value) ? value.key : value.property;
+
+  if (t.isStringLiteral(keyOrProperty)) {
+    return keyOrProperty;
+  }
+
+  if (t.isIdentifier(keyOrProperty)) {
+    if (t.isObjectProperty(value) && value.computed) return undefined;
+    return value.computed ? keyOrProperty : t.stringLiteral(keyOrProperty.name);
+  }
+
+  return undefined;
+};
+
+/**
+ * Retrieves the CSS Map related information from a given expression.
+ *
+ * @param expression The expression to be evaluated.
+ * @param meta {Metadata} Useful metadata that can be used during the transformation
+ * @param evaluateExpression {EvaluateExpression} Function that evaluates an expression
+ */
+const getCssMap = (
+  expression: t.Expression,
+  meta: Metadata,
+  evaluateExpression: EvaluateExpression
+):
+  | {
```

Comment:
Nit: not sure what this extra pipe character does here

Reply:
I wonder why the eslint plugin doesn't catch this 🤔

Reply:
believe it or not. The extra pipe is enforced by prettier. 🤷‍♂️

Reply:
TIL